static/js/publisher/form/media.js

```diff
@@ -0,0 +1,225 @@
+import React, { Fragment } from "react";
+import ReactDOM from "react-dom";
+
+import { PropTypes } from "prop-types";
+
+import MediaItem, { mediaClasses } from "./mediaItem";
+
+class Media extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.markForDeletion = this.markForDeletion.bind(this);
+    this.addImage = this.addImage.bind(this);
+    this.mediaChange = this.mediaChange.bind(this);
+    this.keyboardEvent = this.keyboardEvent.bind(this);
+    this.focused = this.focused.bind(this);
+    this.blurred = this.blurred.bind(this);
+
+    this.state = {
+      mediaData: props.mediaData,
+      focused: null
+    };
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    this.props.updateState(prevState.mediaData);
+    if (
+      prevState.mediaData.filter(item => item.status !== "delete").length >
+      this.props.mediaLimit
+    ) {
+      // Do the query lookup only when we need it - otherwise every state
+      // update will trigger a DOM lookup
+      const warning = document.querySelector(".js-media-limit-warning");
```

Comment:
Why toggling external DOM element? Can't warning be rendered by React only if needed?

Reply:
It's in the template for no-js rendering. Didn't want to duplicate the element